analysis/topEFT/topeft.py

```diff
@@ -261,77 +207,63 @@ def process(self, df):
-        # Create pairs
-        eee_groups = eee.distincts()
-        mmm_groups = mmm.distincts()
-        # Calculate the invariant mass of the pairs
-        invMass_eee = ((eee_groups.i0+eee_groups.i1).mass)
-        invMass_mmm = ((mmm_groups.i0+mmm_groups.i1).mass)
-        # OS pairs
-        isOSeee = ((eee_groups.i0.charge != eee_groups.i1.charge))
-        isOSmmm = ((mmm_groups.i0.charge != mmm_groups.i1.charge))
-        # Get the ones with a mass closest to the Z mass (and in a range of  thr)
-        clos_eee = IsClosestToZ(invMass_eee, thr=15)
-        clos_mmm = IsClosestToZ(invMass_mmm, thr=15)
-        # Finally, the mask for eee/mmm with/without OS onZ pair
-        eeeOnZmask  = (clos_eee)&(isOSeee)
-        eeeOffZmask = (eeeOnZmask==0)
-        mmmOnZmask  = (clos_mmm)&(isOSmmm)
-        mmmOffZmask = (mmmOnZmask==0)
-        eeeOnZmask  = (eeeOnZmask[eeeOnZmask].counts>0)
-        eeeOffZmask = (eeeOffZmask[eeeOffZmask].counts>0)
-        mmmOnZmask  = (mmmOnZmask[mmmOnZmask].counts>0)
-        mmmOffZmask = (mmmOffZmask[mmmOffZmask].counts>0)
-        
-        # Get Z and W invariant masses
-        eZ0 = []; eZ1 = []; mZ0 = []; mZ1 = []
-        goodPairs_eee = eee_groups[(clos_eee)&(isOSeee)]
-        goodPairs_mmm = mmm_groups[(clos_mmm)&(isOSmmm)]
-        if len(goodPairs_eee.i0[goodPairs_eee.counts>0])>0:
-          eZ0   = goodPairs_eee.i0[goodPairs_eee.counts>0].regular()
-          eZ1   = goodPairs_eee.i1[goodPairs_eee.counts>0].regular()
-          mZ0   = goodPairs_mmm.i0[goodPairs_mmm.counts>0].regular()
-          mZ1   = goodPairs_mmm.i1[goodPairs_mmm.counts>0].regular()
-
-        eee_reg = eee[(eeeOnZmask)].regular() if len(eee[(eeeOnZmask)])>0 else eee[(eeeOnZmask)]
-        eW = []; mW = []
-        if len(eZ0) > 0:
-          eW = np.append(eee_reg, eZ0, axis=1)
-          eW = np.append(eW, eZ1,axis=1)
-          eWmask = np.apply_along_axis(lambda a : [list(a).count(x)==1 for x in a], 1, eW)
-          eW = eW[eWmask]
-          mmm_reg = mmm[(mmmOnZmask)].regular() if len(mmm[(mmmOnZmask)])>0 else mmm[(mmmOnZmask)]
-          mW = np.append(mmm_reg, mZ0, axis=1)
-          mW = np.append(mW, mZ1,axis=1)
-          mWmask = np.apply_along_axis(lambda a : [list(a).count(x)==1 for x in a], 1, mW)
-          mW = mW[mWmask]
-        
-        eZ      = [x+y for x,y in zip(eZ0, eZ1)]
-        triElec = [x+y for x,y in zip(eZ, eW)]
-        mZ_eee  = [t[0].mass for t in eZ]
-        m3l_eee = [t[0].mass for t in triElec]
-        mZ      = [x+y for x,y in zip(mZ0, mZ1)]
-        triMuon = [x+y for x,y in zip(mZ, mW)]
-        mZ_mmm  = [t[0].mass for t in mZ]
-        m3l_mmm = [t[0].mass for t in triMuon]
-
 
+        ee_pairs = eee.argchoose(2)
+        mm_pairs = mmm.argchoose(2)
+
+        # Select pairs that are SFOS.
+        eeSFOS_pairs = ee_pairs[(np.abs(eee[ee_pairs.i0].pdgId) == np.abs(eee[ee_pairs.i1].pdgId)) & (eee[ee_pairs.i0].charge != eee[ee_pairs.i1].charge)]
```

Comment:
Question: What is the purpose of the check on the equality of pdgId for the electrons in the pair?  Is this enforcing that they're both electrons at MC truth level?